File: gluecodium/src/test/resources/smoke/method_overloads/output/android/com/example/smoke/SpecialNames.java
/*
 *
 */
package com.example.smoke;
import android.support.annotation.NonNull;
import com.example.NativeBase;
public final class SpecialNames extends NativeBase {
    public SpecialNames(@NonNull final String result) {
        this(make(result), (Object)null);
        cacheThisInstance();
    }
    /**
     * For internal use only.
     * @hidden
     */
    protected SpecialNames(final long nativeHandle, final Object dummy) {
        super(nativeHandle, new Disposer() {
            @Override
            public void disposeNative(long handle) {
                disposeNativeHandle(handle);
            }
        });
    }
    private static native void disposeNativeHandle(long nativeHandle);
    private native void cacheThisInstance();
    public native void create();
    public native void release();
    public native void createProxy();
    public native void Uppercase();
    private static native long make(@NonNull final String result);
}

File: gluecodium/src/test/resources/smoke/method_overloads/output/android-kotlin/com/example/smoke/SpecialNames.kt
/*

 *
 */

@file:JvmName("SpecialNamesExtensions")

package com.example.smoke

import com.example.NativeBase

class SpecialNames : NativeBase {




    constructor(result: String) : this(make(result), null as Any?) {
        cacheThisInstance();
    }

    /**
     * For internal use only.
     * @suppress
     * @param nativeHandle The handle to resources on C++ side.
     * @param tag Tag used by callers to avoid overload resolution problems.
     */
    protected constructor(nativeHandle: Long, @Suppress("UNUSED_PARAMETER") tag: Any?)
        : super(nativeHandle, { disposeNativeHandle(it) }) {}

    private external fun cacheThisInstance()




    external fun create() : Unit


    external fun release() : Unit


    external fun createProxy() : Unit


    external fun Uppercase() : Unit




    companion object {
        @JvmStatic private external fun disposeNativeHandle(nativeHandle: Long)

        @JvmStatic external fun make(result: String) : Long
    }
}

